doc/contributing/traitsInDepth.md

```diff
@@ -0,0 +1,45 @@
+# Traits in Depth
+
+The Trait system is Terria's way of managing a priority order for state. For example, we might have a default opacity property from a config file that should be overriden by a user provided value. This article will go through how Traits are made, how the priority system works and how updates happen in the UI.
+
+## Making a Trait
+![Making a trait](./img/making_a_trait.png)
+
+For base traits such as `BaseMapsTraits`, we extend ModelTraits. However when composing multiple traits to form a new trait, we use the `mixTraits` function. For example, `ArcGisTerrainCatalogItemTraits` is composed out of `UrlTraits, MappableTraits, CatalogMemberTraits`.
+
+When creating a new property within a Trait, we use [decorators](https://babeljs.io/docs/en/babel-plugin-proposal-decorators) to ensure that the field takes part in the Trait system.
+
+
+## Stratum order model
+![Stratum order model](./img/stratum.png)
+Stratums are how Terria determines which value a Trait should resolve to. There are 5 categories of Stratum:
+
+1. Defaults (lowest priority)
+2. Underride
+3. Definition
+4. Override
+5. User (highest priority)
+
+### Default Stratum
+This is the lowest priority stratum best thought of as a sensible fallback value.
+
+### Underride Stratum
+When loading data from external sources, we might want to set values at a lower priority. For example, if a field exists both at a group level and at an individual member level. We might want to set the value to underride at the group level so that the value can be overriden at the individual member level.
+
+### Definition Stratum
+This is the default priority level we should use for values loaded from configuration whether it be from a config file or from an asynchronous call to an external API.
```

Comment:
This is not entire true, traits from asynchronous calls are usually put into their own LoadableStratum.
`LoadableStratum` sit between Definition and Override

Reply:
It might be worth adding a Loadable Stratum section with a description like this

>Load Strata: (singular: Load Stratum) Strata that provide values for a subset of the properties defined on model's traits class, usually by loading them from an external source such as WMS GetCapabilities service. Example: GetCapabilitiesStratum.

From https://github.com/TerriaJS/terriajs/blob/main/doc/contributing/model-layer.md#types-of-classes-in-the-model-layer

Reply:
There can be multiple Load Strata - for example CKAN reference that resolved to a WMS item will have the following strata:

- Default
- Underride
- CkanDatasetStratum
- (WMS)GetCapabilitiesStratum
- Definition
- Override
- User

Reply:
oh what's the difference between Definition and Load? I unfortunately missed Loadable Stratum as I was only looking at the constructor but i see it in the priority numbers.

Reply:
yup, I'll add in the Loadable Stratum in the docs/images. I'll probably abstract out the different types of LoadableStratum though

Reply:
Definition is mostly just configuration from JSON initialisation (catalog) files

Reply:
The usage is quite inconsistent thought 😢 

I would say that Definition should **only** be set through initialisation files.

And Underride should be used when models are created dynamically (eg WebMapServiceCatalogGroup creating WebMapServiceCatalogItems)

Reply:
Another point about Defaults, these are set in trait definitions